internal/clients/team5/forage.go

```diff
@@ -0,0 +1,310 @@
+	deerHunters := int(0) // Number of hunters
+	fishHunters := int(0)
+	probDeerHunting := float64(0.1) // Base probaility to add some randomness
+	probFishHunting := float64(0.1) // So we dont always go for the best type
+
+	for forageType, FOutcome := range forageHistory { // For the whole foraging history
+		for _, returns := range FOutcome {
+			if forageType == shared.DeerForageType && returns.turn == c.gameState().Turn-1 { // If someone foraged for deer last turn
+				deerHunters++          // Increment number of hunters
+				probDeerHunting += 0.1 // Incremenet the probabilty we hunt
+			} else if forageType == shared.FishForageType && returns.turn == c.gameState().Turn-1 {
+				fishHunters++
+				probFishHunting += 0.1
+			}
+		}
+	}
+	c.Logf("[Debug] Number of Deer Hunters from pervious turn %v", deerHunters)
+	c.Logf("[Debug] Number of Fish Hunters %v", fishHunters)
+	if bestForagingMethod == shared.FishForageType { // Fishing is best but 3 Deer hunters last turn
+		bDeer := distuv.Bernoulli{P: 1 - probDeerHunting}     // P(1)[Fishing]=0.6 (1-0.1+0.3*3) if 3 deer hunter
+		bestForagingMethod *= shared.ForageType(bDeer.Rand()) // Multipy the 0 in if Deer Hunting was picked in randomness
+	} else if bestForagingMethod == shared.DeerForageType { // Deer hunting is the best choice but 3 Fishers
+		bFish := distuv.Bernoulli{P: probFishHunting}         // P(1)[Fishing]= 0.1 + 0.1*3 = 0.4
+		bestForagingMethod += shared.ForageType(bFish.Rand()) // +1 [makes it fishing] if Fishing was picked in randomness
+	} // If best foraging was none of the 2 above then return shared.ForageType(-1)
+	// ================================================================
+
+	return bestForagingMethod
+}
+
+/* normalForage() (Past the initial, based on the history of our foraging and some randomness) */
+func (c *client) normalForage() shared.ForageDecision {
+	bestForagingMethod := c.bestHistoryForaging(c.forageHistory) // Find the best foragine type in based on history
+
+	// No good returns all our history had RoI < 0
+	//=============================================================================
+	if bestForagingMethod == shared.ForageType(-1) && c.config.SkipForage > 0 {
+		c.Logf("[Debug] - Skipping Foraging %v", c.config.SkipForage)
+		c.config.SkipForage = c.config.SkipForage - 1 // Count down the number of turns to skip
+		return shared.ForageDecision{                 // Dont go foraging
+			Type:         shared.FishForageType,
+			Contribution: 0,
+		}
+	} else if bestForagingMethod == shared.ForageType(-1) && c.config.SkipForage == 0 { // Force Foraging
+		c.Logf("[Debug] - Force Foraging %v", c.config.SkipForage)
+		c.config.SkipForage = 1 // Reassign the number of skips for next time we have RoI < 0
+
+		// Randomly pick type and invest 1->3%
+		var forageMethod shared.ForageType
+		forageContribution := shared.Resources(0.01+rand.Float64()*(0.03-0.01)) * c.gameState().ClientInfo.Resources
```

Comment:
maybe we should define these fractions as parameters in a config file that we can easily change/monitor